### PR TITLE
[CHAT-1471] Allowed/blocked file extensions and content types

### PR DESCRIPTION
**Pull request by miagilepner**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1251,12 +1253,24 @@ export type AppSettings = {
   disable_auth_checks?: boolean;
   disable_permissions_checks?: boolean;
   enforce_unique_usernames?: 'no' | 'app' | 'team';
+  file_upload_config?: {
+    allowed_file_extensions?: string[];
+    allowed_mime_types?: string[];
```
Comment (vishalnarkhede) [inline src/types.ts:1258]: Just to confirm, do we have some exhaustive list of all these types/extensions? @miagilepner

File: helloworld.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Hello World!
```
Comment (vishalnarkhede) [inline helloworld.md:1]:
```suggestion
```

File: src/types.ts (hunk 1 above)
Comment (miagilepner) [reply src/types.ts:1258]: We don't. We could make an exhaustive list (e.g. for the media types the full list is here: https://www.iana.org/assignments/media-types/media-types.xhtml) and do validation, if you think this should be prioritized.

Comment (vishalnarkhede) [reply src/types.ts:1258]: Not necessary tbh. If backend is prepared to handle any wierd type ... then its totally fine!!

**Review by vishalnarkhede — APPROVED** [review]
Lets wait for @nhannah 's  review as well :)

File: src/types.ts (hunk 1 above)
Comment (mahboubii) [reply src/types.ts:1258]: let's put a comment in here with the link to the accepted types

**Review by ferhatelmas — APPROVED** [review]
LGTM (for tests)

File: src/types.ts [hunk 3]
```diff
@@ -1460,6 +1475,13 @@ export type Field = {
   value?: string;
 };
 
+export type FileUploadConfig = {
```
Comment (ferhatelmas) [inline src/types.ts:1478]: why don't we use this type above ?

Comment (miagilepner) [reply src/types.ts:1478]: was following the way that it's done for other objects, which repeat fields rather than using the type.

Comment (ferhatelmas) [reply src/types.ts:1478]: any example for a reference ?

Comment (miagilepner) [reply src/types.ts:1478]: Ah, you're right. I thought firebase_config and apn_config did this, but I see now they have slightly different fields in AppSettings. I'll fix that.

Comment (ferhatelmas) [reply src/types.ts:1478]: Yes, they are different slightly but here is exactly same. Thanks for updating.

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Review by nhannah — APPROVED** [review]

**Pull request merged**